pages/visualizations/chaoss/contributors_over_time.py

```diff
@@ -139,64 +138,88 @@ def create_graph(repolist, timer_pings, contribs, interval):
     logging.debug("CONTRIBUTIONS_OVER_TIME_VIZ - START")
     start = time.perf_counter()
 
-    df_cont = pd.DataFrame(results)
-    df_cont["created_at"] = pd.to_datetime(df_cont["created_at"], utc=True, format="%Y-%m-%d")
+    # create dataframe from record data
+    df = pd.DataFrame(results)
+
+    # test if there is data
+    if df.empty:
+        logging.debug("PULL REQUESTS OVER TIME - NO DATA AVAILABLE")
+        return nodata_graph, False, dash.no_update
+
+    # convert to datetime objects with consistent column name
+    df["created_at"] = pd.to_datetime(df["created_at"], utc=True)
+    df.rename(columns={"created_at": "created"}, inplace=True)
+
+    # remove null contrib ids
+    df.dropna(inplace=True)
 
     # create column for identifying Drive by and Repeat Contributors
-    contributors = df_cont["cntrb_id"][df_cont["rank"] == contribs].to_list()
-    df_cont["type"] = np.where(df_cont["cntrb_id"].isin(contributors), "Repeat", "Drive-By")
+    contributors = df["cntrb_id"][df["rank"] == contribs].to_list()
 
-    # reset index to be ready for plotly
-    df_cont = df_cont.reset_index()
+    # dfs for drive by and repeat contributors
+    df_drive_temp = df.loc[~df["cntrb_id"].isin(contributors)]
```

Comment:
that's clever.

Reply:
but wouldn't contributors w/ more contributions than contribs also be filtered out? maybe on line 157 it should be:
contributors = df["cntrb_id"][df["rank"] >= contribs].to_list() 

not sure if this is the case

Reply:
It handles that case. An example would be is that say the contrib = 5. Even if a single contributor has 100s of contributions, it will still have one where rank = 5. So the id will be added to the list and all contributions for that id will be included